seive.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+function seiveOriginal(n) {
+  let primes = new Array(n - 1).fill(true);
+  [primes[0], primes[1]] = [false, false];
+  for (let i = 2; i <= n; i++) {
+    for (let j = i * i; j <= n; j += i) {
+      primes[j] = false;
+    }
+  }
+
+  primes.forEach((el, i) => {
+    if (el) console.log(i);
+  });
+}
+
+function blocksSeive(n) {
+  let nPrime = new Array(n).fill(false);
+  let primes = [];
+  const sqrt = Math.sqrt(n);
+  for (let i = 2; i < sqrt; i++) {
```

Comment:
Try to modify the `seiveOriginal` function to return the boolean array and reuse `seiveOriginal` here. This way the code will be more structured.

Reply:
Could the function `seiveOriginal` return the `primes` array?

Reply:
Yeah, that works. There are cases when you want to have a boolean array to check whether a number is prime in constant time. But here it's fine just returning the primes array.